### ISSUE TITLE
spiceweasel barfs when cookbook doesn't have a name ...

### ISSUE THREAD

**Issue by milligana** [chef-boneyard/openstack-chef-repo#2]
$ spiceweasel infrastructure.yml 
ERROR: Invalid version '1.4.2' of 'apache2' requested, '1.3.2' is already in the cookbooks directory.

Seems a lot of cookbooks don't have an explicit name in metadata.rb which causes the above.  Perhaps the top-level directory should be assumed as name if not present ...

Specific gem versions:
$ rpm -qa | grep rubygem- | sort
rubygem-abstract-1.0.0-2.lbn13.ruby193.noarch
rubygem-activemodel-3.2.3-1.lbn13.ruby193.noarch
rubygem-activesupport-3.2.3-1.lbn13.ruby193.noarch
rubygem-addressable-2.3.5-1.lbn13.noarch
rubygem-akami-1.0.0-1.lbn13.ruby193.noarch
rubygem-alchemist-0.1.2.1-1.lbn13.ruby193.noarch
rubygem-archive-tar-minitar-0.5.2-1.lbn13.ruby193.noarch
rubygem-arr-pm-0.0.8-1.lbn13.noarch
rubygem-autotest-4.4.6-1.lbn13.ruby193.noarch
rubygem-aws-sdk-1.6.9-1.lbn13.ruby193.noarch
rubygem-backports-3.3.3-1.lbn13.noarch
rubygem-bacon-1.1.0-3.lbn13.ruby193.noarch
rubygem-berkshelf-2.0.8-2.lbn13.noarch
rubygem-bigdecimal-1.1.0-24.lbn13.ruby193.x86_64
rubygem-bouncy-castle-java-1.5.0146.1-1.lbn13.ruby193.noarch
rubygem-boxgrinder-bastion-1.0.6-1.lbn13.noarch
rubygem-boxgrinder-build-0.10.4-2.lbn13.noarch
rubygem-boxgrinder-core-0.3.14-1.lbn13.ruby193.noarch
rubygem-buff-config-0.4.0-1.lbn13.noarch
rubygem-buff-extensions-0.5.0-1.lbn13.noarch
rubygem-buff-ignore-1.1.0-1.lbn13.noarch
rubygem-buff-ruby_engine-0.1.0-1.lbn13.noarch
rubygem-buff-shell_out-0.1.0-1.lbn13.noarch
rubygem-builder-3.0.0-1.lbn13.ruby193.noarch
rubygem-bundler-1.2.1-1.lbn13.ruby193.noarch
rubygem-bunny-0.8.0-1.lbn13.ruby193.noarch
rubygem-cabin-0.6.0-1.lbn13.noarch
rubygem-celluloid-0.14.0-1.lbn13.noarch
rubygem-celluloid-io-0.14.0-1.lbn13.noarch
rubygem-chef-zero-1.5.5-1.lbn13.noarch
rubygem-childprocess-0.3.5-1.lbn13.ruby193.noarch
rubygem-chozo-0.6.1-1.lbn13.noarch
rubygem-chronic-0.6.7-1.lbn13.ruby193.noarch
rubygem-clamp-0.6.0-1.lbn13.noarch
rubygem-coderay-1.0.8-1.lbn13.ruby193.noarch
rubygem-coffee-script-source-1.2.0-1.lbn13.ruby193.noarch
rubygem-configparser-0.1.0-1.lbn13.ruby193.noarch
rubygem-configuration-1.3.2-1.lbn13.ruby193.noarch
rubygem-contest-0.1.3-1.lbn13.ruby193.noarch
rubygem-cookiejar-0.3.0-1.lbn13.ruby193.noarch
rubygem-crack-0.1.8-1.lbn13.ruby193.noarch
rubygem-creole-0.4.2-1.lbn13.ruby193.noarch
rubygem-curb-0.7.15-1.lbn13.ruby193.x86_64
rubygem-delorean-2.0.0-1.lbn13.ruby193.noarch
rubygem-diff-lcs-1.1.3-1.lbn13.ruby193.noarch
rubygem-em-http-request-1.0.3-1.lbn13.ruby193.noarch
rubygem-em-socksify-0.2.1-1.lbn13.ruby193.noarch
rubygem-erubis-2.7.0-1.lbn13.ruby193.noarch
rubygem-eventmachine-1.0.0-1.lbn13.ruby193.x86_64
rubygem-excon-0.25.3-1.lbn13.noarch
rubygem-expression_parser-0.9.0-1.lbn13.ruby193.noarch
rubygem-faraday-0.8.8-1.lbn13.noarch
rubygem-ffi-1.9.0-1.lbn13.x86_64
rubygem-fiber-0.1-1.lbn13.ruby193.noarch
rubygem-fog-1.14.0-1.lbn13.noarch
rubygem-formatador-0.2.3-1.lbn13.ruby193.noarch
rubygem-fpm-0.4.42-1.lbn13.noarch
rubygem-ftw-0.0.34-1.lbn13.noarch
rubygem-gem2rpm-0.9.2-1.lbn13.noarch
rubygem-gemcutter-0.7.1-1.lbn13.ruby193.noarch
rubygem-gettext-2.2.1-1.lbn13.ruby193.noarch
rubygem-git-1.2.5-1.lbn13.ruby193.noarch
rubygem-glib2-0.90.5-3.lbn13.ruby193.x86_64
rubygem-gssapi-1.0.0-1.lbn13.ruby193.noarch
rubygem-gyoku-0.4.4-1.lbn13.ruby193.noarch
rubygem-haml-3.1.4-1.lbn13.ruby193.noarch
rubygem-hashie-2.0.5-1.lbn13.noarch
rubygem-hiera-1.1.2-2.lbn13.noarch
rubygem-highline-1.6.12-1.lbn13.ruby193.noarch
rubygem-hoe-3.0.2-1.lbn13.ruby193.noarch
rubygem-hpricot-0.8.4-1.lbn13.ruby193.x86_64
rubygem-httparty-0.9.0-1.lbn13.ruby193.noarch
rubygem-httpclient-2.2.1-1.lbn13.ruby193.x86_64
rubygem-httpi-0.9.5-1.lbn13.ruby193.noarch
rubygem-http_parser.rb-0.5.3-1.lbn13.ruby193.x86_64
rubygem-i18n-0.6.0-1.lbn13.ruby193.noarch
rubygem-io-console-0.3-24.lbn13.ruby193.x86_64
rubygem-ipaddress-0.8.0-1.lbn13.ruby193.noarch
rubygem-jeweler-1.8.4-1.lbn13.ruby193.noarch
rubygem-jruby-openssl-0.7.6.1-1.lbn13.ruby193.noarch
rubygem-json-1.7.7-100.lbn13.x86_64
rubygem-json_pure-1.6.3-7.lbn13.noarch
rubygem-knife-backup-0.0.7-2.lbn13.noarch
rubygem-knife-ec2-0.6.4-1.lbn13.noarch
rubygem-knife-essentials-1.5.3-2.lbn13.noarch
rubygem-knife-kvm-0.2-1.lbn13.ruby193.noarch
rubygem-kramdown-0.13.5-1.lbn13.ruby193.noarch
rubygem-kwalify-0.7.2-1.lbn13.ruby193.noarch
rubygem-launchy-2.1.2-1.lbn13.ruby193.noarch
rubygem-librarian-0.1.1-1.lbn13.noarch
rubygem-librarian-chef-0.0.1-1.lbn13.noarch
rubygem-libv8-3.11.8.13-1.lbn13.x86_64
rubygem-liquid-2.3.0-1.lbn13.ruby193.noarch
rubygem-little-plugger-1.1.3-1.lbn13.noarch
rubygem-locale-2.0.5-1.lbn13.ruby193.1.noarch
rubygem-log4r-1.1.10-1.lbn13.ruby193.noarch
rubygem-logging-1.8.0-3.lbn13.noarch
rubygem-markaby-0.7.2-2.lbn13.ruby193.noarch
rubygem-maruku-0.6.1-1.lbn13.ruby193.noarch
rubygem-mime-types-1.23-1.lbn13.noarch
rubygem-minitar-0.5.4-1.lbn13.ruby193.noarch
rubygem-minitest-3.3.0-1.lbn13.ruby193.noarch
rubygem-mixlib-authentication-1.3.0-1.lbn13.ruby193.noarch
rubygem-mixlib-cli-1.2.2-1.lbn13.ruby193.noarch
rubygem-mixlib-config-1.1.2-1.lbn13.ruby193.noarch
rubygem-mixlib-log-1.4.1-1.lbn13.ruby193.noarch
rubygem-mixlib-shellout-1.1.0-3.lbn13.ruby193.noarch
rubygem-mocha-0.12.3-1.lbn13.ruby193.noarch
rubygem-moneta-0.6.0-1.lbn13.ruby193.noarch
rubygem-multi_json-1.3.6-1.lbn13.ruby193.noarch
rubygem-multipart-post-1.2.0-1.lbn13.noarch
rubygem-multi_xml-0.5.1-1.lbn13.ruby193.noarch
rubygem-mustache-0.99.4-1.lbn13.ruby193.noarch
rubygem-net-http-persistent-2.8-1.lbn13.ruby193.noarch
rubygem-net-scp-1.1.2-1.lbn13.noarch
rubygem-net-sftp-2.0.5-2.lbn13.ruby193.noarch
rubygem-net-ssh-2.6.5-1.lbn13.noarch
rubygem-net-ssh-gateway-1.1.0-2.lbn13.ruby193.noarch
rubygem-net-ssh-multi-1.1-2.lbn13.ruby193.noarch
rubygem-nio4r-0.5.0-1.lbn13.x86_64
rubygem-nokogiri-1.5.5-1.lbn13.ruby193.x86_64
rubygem-nori-1.0.2-1.lbn13.ruby193.noarch
rubygem-ohai-6.18.0-1.lbn13.noarch
rubygem-omnibus-1.2.0-1.lbn13.noarch
rubygem-omnibus-software-0.0.1-1.lbn13.noarch
rubygem-open4-1.3.0-1.lbn13.ruby193.noarch
rubygem-patron-0.4.18-1.lbn13.x86_64
rubygem-Platform-0.4.0-1.lbn13.ruby193.noarch
rubygem-polyglot-0.3.3-2.lbn13.ruby193.noarch
rubygem-popen4-0.1.2-1.lbn13.ruby193.noarch
rubygem-progressbar-0.11.0-1.lbn13.ruby193.noarch
rubygem-puma-1.6.3-1.lbn13.x86_64
rubygem-rack-1.5.2-1.lbn13.noarch
rubygem-rack-protection-1.3.2-1.lbn13.noarch
rubygem-rack-test-0.6.2-1.lbn13.ruby193.noarch
rubygem-radius-0.7.3-1.lbn13.ruby193.noarch
rubygem-rake-10.1.0-1.lbn13.noarch
rubygem-rake-compiler-0.8.1-1.lbn13.ruby193.noarch
rubygem-rb-readline-0.4.1-1.lbn13.ruby193.noarch
rubygem-rbzip2-0.2.0-1.lbn13.noarch
rubygem-rdiscount-1.6.8-1.lbn13.ruby193.x86_64
rubygem-rdoc-3.12-2.lbn13.ruby193.noarch
rubygem-redcarpet-2.1.1-1.lbn13.ruby193.x86_64
rubygem-RedCloth-4.2.2-1.lbn13.ruby193.x86_64
rubygem-rest-client-1.6.7-1.lbn13.ruby193.noarch
rubygem-retryable-1.3.3-1.lbn13.noarch
rubygem-ridley-1.5.2-1.lbn13.noarch
rubygem-ronn-0.7.3-1.lbn13.ruby193.noarch
rubygem-rspec-2.11.0-1.lbn13.ruby193.noarch
rubygem-rspec-core-2.11.1-2.lbn13.ruby193.noarch
rubygem-rspec-expectations-2.11.2-1.lbn13.ruby193.noarch
rubygem-rspec-mocks-2.11.2-1.lbn13.ruby193.noarch
rubygem-rubyforge-2.0.4-2.lbn13.ruby193.noarch
rubygem-ruby-hmac-0.4.0-2.lbn13.ruby193.noarch
rubygem-rubyntlm-0.1.1-1.lbn13.ruby193.noarch
rubygem-sass-3.1.15-2.lbn13.ruby193.noarch
rubygem-savon-0.9.6-1.lbn13.ruby193.noarch
rubygem-shindo-0.3.4-1.lbn13.ruby193.noarch
rubygem-shoulda-2.11.3-1.lbn13.ruby193.noarch
rubygem-sinatra-1.3.3-2.lbn13.ruby193.noarch
rubygem-solve-0.8.0-1.lbn13.noarch
rubygem-spiceweasel-2.5.1-1.lbn13.noarch
rubygem-syntax-1.0.0-3.lbn13.ruby193.noarch
rubygem-SyslogLogger-2.0-1.lbn13.noarch
rubygem-systemu-2.5.2-1.lbn13.ruby193.noarch
rubygem-term-ansicolor-1.0.7-1.lbn13.ruby193.noarch
rubygem-terminal-table-1.4.5-1.lbn13.ruby193.noarch
rubygem-test-unit-2.2.0-1.lbn13.ruby193.noarch
rubygem-therubyrhino-1.73.1-1.lbn13.ruby193.noarch
rubygem-thor-0.18.1-1.lbn13.noarch
rubygem-tidy-ext-0.1.12-1.lbn13.ruby193.x86_64
rubygem-tilt-1.3.3-3.lbn13.ruby193.noarch
rubygem-timecop-0.3.5-2.lbn13.ruby193.noarch
rubygem-timers-1.1.0-1.lbn13.noarch
rubygem-treetop-1.4.10-1.lbn13.ruby193.noarch
rubygem-uber-s3-0.2.4-1.lbn13.noarch
rubygem-uuidtools-2.1.3-1.lbn13.ruby193.noarch
rubygem-vagrant-1.0.5-1.lbn13.ruby193.noarch
rubygem-varia_model-0.2.0-1.lbn13.noarch
rubygem-wasabi-2.0.0-1.lbn13.ruby193.noarch
rubygem-webmock-1.6.4-1.lbn13.ruby193.noarch
rubygem-winrm-1.1.2-1.lbn13.noarch
rubygem-yajl-ruby-1.1.0-2.lbn13.ruby193.x86_64
rubygem-yard-0.8.7-1.lbn13.noarch
rubygem-ZenTest-4.8.2-1.lbn13.ruby193.noarch

**Comment by milligana** [comment]
Sorry - the above was using the wrong knife config - darn...  Versions same, but here's the message:

spiceweasel -c .chef/knife.rb.save infrastructure.yml
ERROR: Cookbook 'openssl' does not match the name 'openssl' in openssl/metadata.rb.

**Comment by mattray** [comment]
This repo is out of date and will be updated soon. If you want to stay on the latest community efforts, look over here:
https://github.com/stackforge/openstack-chef-repo

**Comment by milligana** [comment]
Matt,

Thanks for that, I'll look into it.

However, the spiceweasel name issue still stands.  A large number of 
recipes don't have a "name" explicitly set in metadata.rb (eg openssl, 
sysctl,...).

I also discovered something nasty which I believe spiceweasel did: on 
the execute step, it blew away _all_ the workspace copies of _all_ the 
cookbooks not listed in the yaml.  I had a lot of other development 
shite there that I had to recover from the server directly :(   Whatever 
happened to the Hypocratic Oath and "do no harm" ...

Alan
